wp-content/themes/engage/functions.php

```diff
@@ -51,6 +51,31 @@
 	acf_add_options_page();
 }
 
+add_filter( 'pre_get_posts', 'tribe_change_event_order', 99 );
+
+// When all previous events are viewable in one page, the events will
+// be sorted from most recent to oldest
+function tribe_change_event_order( $query ) {
+    if ( $query->get( 'posts_per_page' ) == -1 && tribe_is_past() ) {
+        $query->set( 'orderby', 'date' );
+        $query->set( 'order', 'ASC' );
+        add_filter( 'tribe_get_events_title', 'tribe_alter_event_archive_titles', 11, 2 );
+    }
+
+    return $query;
+}
+
+function tribe_alter_event_archive_titles( $original_recipe_title, $depth ) {
```

Comment:
@leoncai1999 Do you think this function is creating the title bug you mentioned?